src/App.js

```diff
@@ -29,37 +29,114 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+  const [winner, setWinner] = useState(null);
+  const [moves, setMoves] = useState(0);
 
   // Wave 2
   // You will need to create a method to change the square
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
+  const updateSquares = (id) => {
+    const updatedSquares = [...squares];
+    for (let i = 0; i < 3; i++) {
+      for (let j = 0; j < 3; j++) {
+        if (
+          updatedSquares[i][j].id === id &&
+          updatedSquares[i][j].value === '' &&
+          winner === null
+        ) {
+          updatedSquares[i][j].value = currentPlayer;
+          changeCurrentPlayer();
+          setMoves(moves + 1);
+        }
+      }
+    }
+    setSquares(updatedSquares);
+    setWinner(checkForWinner);
+  };
+
+  const changeCurrentPlayer = () => {
+    if (currentPlayer === PLAYER_1) {
+      setCurrentPlayer(PLAYER_2);
+    } else {
+      setCurrentPlayer(PLAYER_1);
+    }
```

Comment:
if we wanna get super fancy, we could turn this into a one line ternary operator!